experimental/streaming/exporter/reader/reader.go

```diff
@@ -266,22 +255,20 @@ func (ro *readerObserver) orderedObserve(event observer.Event) {
 			read.SpanContext = span.spanContext
 		}
 
-	case observer.RECORD_STATS:
-		read.Type = RECORD_STATS
+	case observer.UPDATE_METRIC:
 
-		_, span := ro.readScope(event.Scope)
-		if span != nil {
-			read.SpanContext = span.spanContext
-		}
-		for _, es := range event.Stats {
-			ro.addMeasurement(&read, es)
-		}
-		if event.Stat.Measure != nil {
-			ro.addMeasurement(&read, event.Stat)
+		read.Type = event.Type
+
+		if event.Context != nil {
+			span := trace.CurrentSpan(event.Context)
```

Comment:
Here, this is where the metric update is associated with a current span. This would be a `nil` context if for some reason the programmer passed a nil to the metrics APIs.